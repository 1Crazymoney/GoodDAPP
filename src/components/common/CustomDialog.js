// @flow
import React from 'react'
import { Portal, Dialog, Paragraph } from 'react-native-paper'
import CustomButton from './CustomButton'

type DialogProps = {
  children?: any,
  visible?: boolean,
  title?: string,
  message?: string,
  dismissText?: string,
  onDismiss?: () => void,
  loading?: boolean
}

const CustomDialog = ({ children, visible, title, message, dismissText, onDismiss, loading = false }: DialogProps) => (
  <Portal>
    <Dialog visible={visible} onDismiss={onDismiss} dismissable={true}>
      <Dialog.Title>{title}</Dialog.Title>
      <Dialog.Content>
        {children ? children : null}
        {message ? <Paragraph>{message}</Paragraph> : null}
      </Dialog.Content>
      <Dialog.Actions>
        <CustomButton onPress={onDismiss} disabled={loading} loading={loading}>
          {dismissText || 'Done'}
        </CustomButton>
      </Dialog.Actions>
    </Dialog>
  </Portal>
)

export default CustomDialog
