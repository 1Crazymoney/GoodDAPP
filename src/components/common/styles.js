import { normalize } from 'react-native-elements'

export const fontStyle = {
  color: '#555',
  fontSize: normalize(18),
  textAlign: 'center',
  fontFamily: 'Roboto'
}

export const scrollableContainer = {
  minHeight: '100%'
}
