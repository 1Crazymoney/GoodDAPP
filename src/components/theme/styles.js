import DefaultTheme from 'react-native-paper/src/styles/DefaultTheme'
import normalize from 'react-native-elements/src/helpers/normalizeText'

export const theme = {
  ...DefaultTheme,
  borders: {
    defaultBorderColor: '#c9c8c9',
  },
  colors: {
    ...DefaultTheme.colors,
    blue: '#006EA0',
    darkBlue: '#0C263D',
    darkGray: '#42454A',
    gray50Percent: '#CBCBCB',
    placeholder: '#CBCBCB',
    green: '#00C3AE',
    lightGray: '#EEE',
    lighterGray: '#A3A3A3',
    orange: '#F8AF40',
    primary: '#00AFFF',
    purple: '#9F6A9D',
    red: '#FA6C77',
    text: '#222',
  },
  fonts: {
    ...DefaultTheme.fonts,
    slab: 'RobotoSlab-Regular',
    slabBold: 'RobotoSlab-Bold',
    bold: 'Roboto-Bold',
  },
  paddings: {
    mainContainerPadding: normalize(8),
    defaultMargin: normalize(8),
  },
  sizes: {
    default: normalize(8),
    defaultDouble: normalize(16),
    defaultHalf: normalize(4),
  },
  fontStyle: {
    color: '#555',
    fontSize: normalize(18),
    textAlign: 'center',
  },
  modals: {
    overlayBackgroundColor: 'rgba(255, 255, 255, 0.8)',
    overlayHorizontalPadding: normalize(20),
    overlayVerticalPadding: normalize(30),
    backgroundColor: '#fff',
    jaggedEdgeSize: normalize(15),
    contentPadding: normalize(16),
    borderRadius: normalize(5),
    borderLeftWidth: normalize(10),
  },
  feedItems: {
    borderRadius: normalize(8),
    itemBackgroundColor: '#fff',
    height: normalize(84),
  },
}
