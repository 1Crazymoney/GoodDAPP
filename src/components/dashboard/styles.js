import normalize from 'react-native-elements/src/helpers/normalizeText'
import { StyleSheet } from 'react-native'

import { fontStyle } from '../common/styles'

export const receiveStyles = StyleSheet.create({
  wrapper: {
    justifyContent: 'flex-start',
    width: '100%',
    padding: normalize(8)
  },
  section: {
    flex: 1
  },
  sectionRow: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%'
  },
  qrCode: {
    marginTop: '2rem',
    padding: '1rem',
    borderColor: '#555',
    borderWidth: 1,
    borderRadius: '4px'
  },
  addressSection: {
    marginBottom: '1rem'
  },
  address: {
    margin: '0.5rem'
  },
  secondaryText: {
    margin: '1rem',
    color: '#555555',
    fontSize: normalize(14),
    textTransform: 'uppercase'
  },
  lowerSecondaryText: {
    margin: '1rem',
    color: '#555555',
    fontSize: normalize(16)
  },
  headline: {
    ...fontStyle,
    textTransform: 'uppercase',
    marginBottom: '1rem',
    fontSize: normalize(24)
  },
  buttonGroup: {
    width: '100%',
    flexDirection: 'row',
    marginTop: '1rem'
  },
  inputField: {
    width: '100%',
    flexDirection: 'column',
    justifyContent: 'space-between'
  },
  amountLabel: {
    ...fontStyle,
    fontSize: normalize(32)
  },
  amountSymbol: {
    fontSize: normalize(12)
  },
  amountWrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignContent: 'center'
  },
  amountInput: {
    backgroundColor: 'transparent',
    height: normalize(40),
    width: '100%'
  },
  amountInputWrapper: {
    fontSize: normalize(26),
    lineHeight: normalize(40),
    whiteSpace: 'normal',
    flexShrink: 1,
    flexGrow: 1,
    textAlign: 'right'
  },
  shareQRButton: {
    width: '100%'
  },
  doneButton: {
    marginTop: '1rem'
  },
  fullWidth: {
    marginHorizontal: normalize(10)
  },
  amountSuffix: {
    flexGrow: 1,
    height: normalize(40),
    fontSize: normalize(10),
    justifyContent: 'center',
    lineHeight: normalize(40),
    paddingTop: normalize(10)
  },
  buttonStyle: {
    marginTop: '1em'
  }
})
