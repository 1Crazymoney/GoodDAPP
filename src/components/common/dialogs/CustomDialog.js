// @flow
import React from 'react'
import { StyleSheet, View } from 'react-native'
import { Dialog, Paragraph, Portal } from 'react-native-paper'
import normalize from 'react-native-elements/src/helpers/normalizeText'
import SimpleStore from '../../../lib/undux/SimpleStore'
import CustomButton from '../buttons/CustomButton'
import ModalCloseButton from '../modal/ModalCloseButton'
import ModalLeftBorder from '../modal/ModalLeftBorder'
import ModalContents from '../modal/ModalContents'
import ModalInnerContents from '../modal/ModalInnerContents'
import ModalContainer from '../modal/ModalContainer'
import { theme } from '../../theme/styles'

export type DialogProps = {
  children?: any,
  dismissText?: string,
  image?: any,
  loading?: boolean,
  message?: string,
  onCancel?: () => void,
  onDismiss?: () => void,
  title?: string,
  type?: string,
  visible?: boolean,
}

/**
 * Custom Dialog based on react-native-paper
 * @param {DialogProps} props
 * @param {React.Node|String} [props.children]
 * @param {function} [props.onDismiss]
 * @param {function} [props.onCancel]
 * @param {boolean} [props.visible]
 * @param {string} [props.title]
 * @param {string} [props.message]
 * @param {string} [props.dismissText]
 * @param {boolean} [props.loading]
 * @returns {React.Node}
 */
const CustomDialog = ({
  children = null,
  dismissText,
  image,
  loading = false,
  message = null,
  onCancel = null,
  onDismiss,
  title,
  type = 'common',
  visible,
}: DialogProps) => (
  <Portal>
    <Dialog style={styles.clearDialogStyles} visible={visible} onDismiss={onCancel || onDismiss} dismissable={true}>
      {onCancel || onDismiss ? <ModalCloseButton onClose={onCancel || onDismiss} /> : null}
      <ModalContainer>
        <ModalLeftBorder style={{ backgroundColor: getColorFromType(type) }} />
        <ModalContents>
          <ModalInnerContents>
            <Dialog.Title style={styles.title}>{title}</Dialog.Title>
            <View style={styles.content}>
              {children}
              {image ? image : null}
              {message && <Paragraph style={styles.paragraph}>{message}</Paragraph>}
            </View>
            <Dialog.Actions style={styles.buttonsContainer}>
              {onCancel && (
                <CustomButton disabled={loading} loading={loading} onPress={onCancel}>
                  Cancel
                </CustomButton>
              )}
              <CustomButton
                disabled={loading}
                loading={loading}
                onPress={onDismiss}
                style={[styles.buttonOK, { backgroundColor: getColorFromType(type) }]}
              >
                {dismissText || 'Done'}
              </CustomButton>
            </Dialog.Actions>
          </ModalInnerContents>
        </ModalContents>
      </ModalContainer>
    </Dialog>
  </Portal>
)

const getColorFromType = (type: string) => {
  return (
    {
      success: theme.colors.primary,
      error: theme.colors.red,
    }[type] || theme.colors.primary
  )
}

const SimpleStoreDialog = () => {
  const store = SimpleStore.useStore()
  const { dialogData } = store.get('currentScreen')
  return (
    <CustomDialog
      {...dialogData}
      onDismiss={(...args) => {
        const currentDialogData = { ...dialogData }
        store.set('currentScreen')({ dialogData: { visible: false } })
        currentDialogData.onDismiss && currentDialogData.onDismiss(currentDialogData)
      }}
    />
  )
}

const styles = StyleSheet.create({
  clearDialogStyles: {
    backgroundColor: 'transparent',
    padding: 0,
  },
  title: {
    color: theme.colors.darkGray,
    fontFamily: theme.fonts.slabBold,
    fontSize: normalize(24),
    marginBottom: normalize(16),
    textAlign: 'center',
  },
  paragraph: {
    color: theme.colors.darkGray,
    fontSize: normalize(16),
    textAlign: 'center',
  },
  content: {
    display: 'flex',
    flexDirection: 'column',
    flexGrow: 1,
    marginBottom: normalize(40),
    padding: 0,
  },
  buttonsContainer: {
    paddingLeft: 0,
    paddingRight: 0,
  },
  buttonOK: {
    minWidth: normalize(80),
  },
})

export { CustomDialog as default, SimpleStoreDialog }
