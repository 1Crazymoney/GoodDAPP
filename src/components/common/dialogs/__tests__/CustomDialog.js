import React from 'react'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

import { Provider as PaperProvider } from 'react-native-paper'
import { theme } from '../../../theme/styles'
import CustomDialog from '../CustomDialog'
import { Wrapper } from '../../index'

describe('CustomDialog', () => {
  it('renders without errors', () => {
    const tree = renderer.create(
      <PaperProvider theme={theme}>
        <Wrapper>
          <CustomDialog>Next</CustomDialog>
        </Wrapper>
      </PaperProvider>
    )
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(
      <PaperProvider theme={theme}>
        <Wrapper>
          <CustomDialog>Next</CustomDialog>
        </Wrapper>
      </PaperProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
