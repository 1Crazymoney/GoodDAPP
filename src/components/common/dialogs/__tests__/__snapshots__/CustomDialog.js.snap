// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDialog matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    data-name="viewWrapper"
    style={
      Object {
        "WebkitBoxDirection": "normal",
        "WebkitBoxFlex": 1,
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "0%",
        "WebkitFlexDirection": "column",
        "WebkitFlexGrow": 1,
        "WebkitFlexShrink": 1,
        "backgroundImage": "linear-gradient(to bottom, #00AFFF, #2DC0F7, #28C0EF, #23C0E7, #1EC1DF, #19C1D7, #14C1CF, #0FC2C7, #0FC2C7, #0AC2BF, #05C2B7, #00C3AF)",
        "display": "flex",
        "flexBasis": "0%",
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "msFlexDirection": "column",
        "msFlexNegative": 1,
        "msFlexPositive": 1,
        "msFlexPreferredSize": "0%",
        "paddingBottom": "8px",
        "paddingLeft": "8px",
        "paddingRight": "8px",
        "paddingTop": "8px",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      onResponderGrant={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminationRequest={[Function]}
      onScroll={[Function]}
      onScrollShouldSetResponder={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      onWheel={[Function]}
      style={
        Object {
          "WebkitBoxDirection": "normal",
          "WebkitBoxFlex": 1,
          "WebkitBoxOrient": "vertical",
          "WebkitFlexDirection": "column",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitOverflowScrolling": "touch",
          "WebkitTransform": "translateZ(0px)",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexDirection": "column",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "overflowX": "hidden",
          "overflowY": "auto",
          "transform": "translateZ(0px)",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxFlex": 1,
            "WebkitFlexGrow": 1,
            "display": "flex",
            "flexGrow": 1,
            "height": "100%",
            "msFlexPositive": 1,
          }
        }
      />
    </div>
  </div>
</div>
`;
